src/celeritas/ext/GeantSurfacePhysicsLoader.cc

```diff
@@ -0,0 +1,469 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/ext/GeantSurfacePhysicsLoader.cc
+//---------------------------------------------------------------------------//
+#include "GeantSurfacePhysicsLoader.hh"
+
+#include <algorithm>
+#include <string>
+#include <unordered_map>
+#include <G4LogicalSurface.hh>
+#include <G4OpticalSurface.hh>
+#include <G4Version.hh>
+
+#include "corecel/io/Logger.hh"
+#include "corecel/math/SoftEqual.hh"
+#include "geocel/SurfaceParams.hh"
+
+namespace celeritas
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+// HELPER FUNCTIONS
+//---------------------------------------------------------------------------//
+
+//---------------------------------------------------------------------------//
+/*!
+ * Get a string corresponding to the \c G4SurfaceType selection.
+ */
+char const* to_cstring(G4SurfaceType value)
+{
+#define GSPL_ST_PAIR(ENUMVALUE)              \
+    {                                        \
+        G4SurfaceType::ENUMVALUE, #ENUMVALUE \
+    }
+
+    static std::unordered_map<G4SurfaceType, std::string> const to_cstring_impl
+        = {GSPL_ST_PAIR(dielectric_metal),
+           GSPL_ST_PAIR(dielectric_dielectric),
+           GSPL_ST_PAIR(dielectric_LUT),
+           GSPL_ST_PAIR(dielectric_LUTDAVIS),
+           GSPL_ST_PAIR(dielectric_dichroic),
+           GSPL_ST_PAIR(firsov),
+           GSPL_ST_PAIR(x_ray),
+#if G4VERSION_NUMBER >= 1110
+           GSPL_ST_PAIR(coated)
+#endif
+        };
+    return to_cstring_impl.find(value)->second.c_str();
```

Comment:
I also didn't notice: instead of std::string just use `const char*`. Since these are "static" strings rather than local, the pointers to them will still be valid. Also, formatting will be improved if you add a trailing comma, and finally make sure to check for invalid/unknown enums.
```suggestion

    static std::unordered_map<G4SurfaceType, char const*> const names = {
        GSPL_ST_PAIR(dielectric_metal),
        GSPL_ST_PAIR(dielectric_dielectric),
        GSPL_ST_PAIR(dielectric_LUT),
        GSPL_ST_PAIR(dielectric_LUTDAVIS),
        GSPL_ST_PAIR(dielectric_dichroic),
        GSPL_ST_PAIR(firsov),
        GSPL_ST_PAIR(x_ray),
#if G4VERSION_NUMBER >= 1110
        GSPL_ST_PAIR(coated),
#endif
    };
    if (auto iter = names.find(value); iter != names.end())
    {
        return iter->second;
    }
    return "UNKNOWN";
```